### ISSUE TITLE
Insert image in front of text

### ISSUE THREAD

**Issue by valisirius** [PHPOffice/PHPWord#1087]
Hello,

Please tell me if it is possible to insert an image in front of text, when creating a new docx file.
This feature can be useful.

Thank you.

**Comment by troosan** [comment]
If you want to repeat the image on every page, the only way right now is to put it in the header.

**Comment by valisirius** [comment]
Indeed, but the image in the header is behind the text. Can we create an artifice to put the image in the header in front of the text?
On the other hand I saw that we can put image in every section. but only if we create explicitely a section. Maybe if we coud detect a 'page break' process, we could then create a new section and insert a nea image (with 'wrappingStyle' => 'infront'). Could we do this?

**Comment by troosan** [comment]
We cannot detect when a page break is happening, except if you explicitly put one of course.
And in that case you can of course put the image again. To wrap the text around please check http://phpword.readthedocs.io/en/latest/recipes.html

**Comment by valisirius** [comment]
I have  a document, with a single section, containing a table which can ocupy a page or more pages and I want to put a stamp/image on every page, in the same place, in front of the text from table.
Header allows to put an image on every page, behind the text from table - not OK for me.
Footer allows to put an image on every page, behind the text from table, but in front of the text from footer.
Also I tried to build the final docx from two steps: one step builds a document with a fake header and the second one loads the document and attach the correct stamp in the header, but without result.

**Comment by FBnil** [comment]
@valisirius 
Ok, I got something that is working, but it is based on my branch. I recently added ${myblock/}, this actually is a single element block that searches for the paragraph edges where it is found (useful for bulletpoints).
The trick is naming the image `${myimage/}`, anchor the image to the page and with getBlock() you can get the block, then paste it inside a block on the second page with 
```
	$block = $templateProcessor->getBlock("myimage/");
	$templateProcessor->replaceBlock("page2", $block);
```
This resulted for me in 2 pages with the same image at the same relative page position.

nb: Not necessarily the image name, you can also drop a singleton blockname somewhere inside the description or alternative name, and use that blockname. As long as you see:
```
<pic:cNvPr descr="${mydescription/}" name="${myimage/}" id="1"/>
```

To test it:

Just (temporarily) replace this file:
https://github.com/FBnil/PHPWord/blob/develop/src/PhpWord/TemplateProcessor.php

**Comment by 49167386** [comment]
Insert a shape to the template word. Set the shape text like ${image}. The shape can set  in front of text. Use setImageValue can insert image in front of text.
https://phpword.readthedocs.io/en/latest/templates-processing.html#setimagevalue
